modules/b2b_entities/dlg.c

```diff
@@ -783,12 +783,22 @@ int b2b_prescript_f(struct sip_msg *msg, void *uparam)
 	if(method_value == METHOD_PRACK)
 	{
 		LM_DBG("Received a PRACK - send 200 reply\n");
+
+		// Tracer callback: incoming PRACK and its transaction
+		if (dlg && dlg->tracer_param) // request
+			dlg->tracer_cback(msg, NULL, dlg->tracer_param, B2B_NONE);
+
```

Comment:
@nikbyte , this is wrong here as the `dlg` is not yet matched, is it still NULL. The matching against the `b2b_dlg_t` is done later, after the `search_dialog` label.
Without the `dlg` , I see no way to trace this PRACK.